ext/opcache/jit/zend_jit_trace.c

```diff
@@ -4164,10 +4169,14 @@ static const void *zend_jit_trace(zend_jit_trace_rec *trace_buffer, uint32_t par
 			if ((info & MAY_BE_PACKED_GUARD) != 0
 			 && (trace_buffer->stop == ZEND_JIT_TRACE_STOP_LOOP
 			  || trace_buffer->stop == ZEND_JIT_TRACE_STOP_RECURSIVE_CALL
-			  || trace_buffer->stop == ZEND_JIT_TRACE_STOP_RECURSIVE_RET)
+			  || (trace_buffer->stop == ZEND_JIT_TRACE_STOP_RECURSIVE_RET
+			   && (opline-1)->result_type == IS_VAR
```

Comment:
It's better to keep the conditions exactly the same. We don't have ``IS_VAR`` check above (when we generate type guard). In general, Optimizer may fuse DO_FCALL with the following ASSIGN to IS_CV (I'm not sure if this is enabled or not).

Reply:
Maybe I miss something, but I got this from here:

https://github.com/php/php-src/blob/a99025cb0f2158069c10f0bcbf82d2ece283c423/ext/opcache/jit/zend_jit_trace.c#L4148-L4150

So I'm not sure I understand why the check for IS_VAR is fine there for the type guard, but not for the packed type guard.

Reply:
I see. I looked into the master branch code.
So keep ``IS_VAR`` in PHP-8.3 and remove it in PHP-8.4 and master.